indico/web/client/js/react/components/ReviewRating.jsx

```diff
@@ -0,0 +1,71 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Rating} from 'semantic-ui-react';
+
+import {FinalField, FormFieldAdapter} from 'indico/react/forms';
+
+import './ReviewRating.module.scss';
+
+export default function ReviewRating({min, max, value, disabled, onChange}) {
+  const ratingRange = _.range(min, max + 1);
+  const suiRatingRange = _.range(1, ratingRange.length + 1);
+  const maxRating = suiRatingRange[suiRatingRange.length - 1];
+  const rating = suiRatingRange[ratingRange.findIndex(item => item === value)];
+
+  return (
+    <span styleName="rating-field">
+      <Rating
+        maxRating={maxRating}
+        rating={value === null ? null : rating}
+        disabled={disabled}
+        onRate={(evt, {rating: newRating}) => {
+          // map the value from SUIR Rating to the corresponding value between min and max
+          const val = ratingRange[suiRatingRange.findIndex(item => item === newRating)];
+          onChange(val);
+        }}
+        clearable
+      />
+      <span styleName="rating-value">{value !== null ? value : '-'}</span>
+    </span>
+  );
+}
+
+ReviewRating.propTypes = {
+  min: PropTypes.number,
+  max: PropTypes.number.isRequired,
+  value: PropTypes.number,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func,
+};
+
+ReviewRating.defaultProps = {
+  min: 0,
+  value: null,
+  disabled: false,
+  onChange: null,
+};
+
+function RatingAdapter(props) {
+  return <FormFieldAdapter {...props} as={ReviewRating} undefinedValue={null} />;
+}
+
+export function FinalRating({name, label, ...rest}) {
+  return <FinalField {...rest} name={name} label={label} adapter={RatingAdapter} format={v => v} />;
```

Comment:
I don't think you need a custom adapter here. `component={ReviewRating} undefinedValue={[]}` should be enough.

Reply:
why an empty array?

Reply:
copied from the wrong component. i meant `null`